src/utils/cosmosdb_model_versioned.ts

```diff
@@ -0,0 +1,190 @@
+import {
+  BaseModel,
+  CosmosdbModel,
+  CosmosDecodingError,
+  CosmosErrorResponse,
+  CosmosErrors
+} from "./cosmosdb_model";
+
+import * as t from "io-ts";
+
+import { right } from "fp-ts/lib/Either";
+import { fromNullable, none, Option, some } from "fp-ts/lib/Option";
+import { fromEither, TaskEither, tryCatch } from "fp-ts/lib/TaskEither";
+
+import { PromiseType } from "italia-ts-commons/lib/types";
+
+import { NonNegativeInteger } from "italia-ts-commons/lib/numbers";
+
+import {
+  Container,
+  ErrorResponse,
+  ItemDefinition,
+  RequestOptions,
+  SqlQuerySpec
+} from "@azure/cosmos";
+import { wrapWithKind } from "./types";
+
+export const ModelId = t.string; // FIXME: make it branded
+
+export type ModelId = t.TypeOf<typeof ModelId>;
+
+/**
+ * A VersionedModel should track the version of the model
+ */
+export const VersionedModel = t.interface({
+  version: NonNegativeInteger
+});
+
+export type VersionedModel = t.TypeOf<typeof VersionedModel>;
+
+/**
+ * Returns a string with a composite id that has the format:
+ * MODEL_ID-VERSION
+ *
+ * MODEL_ID is the base model ID
+ * VERSION is the zero-padded version of the model
+ *
+ * @param modelId The base model ID
+ * @param version The version of the model
+ */
+export function generateVersionedModelId(
+  modelId: ModelId,
+  version: NonNegativeInteger
+): string {
+  const paddingLength = 16; // length of Number.MAX_SAFE_INTEGER == 9007199254740991
+  const paddedVersion = ("0".repeat(paddingLength) + String(version)).slice(
+    -paddingLength
+  );
```

Comment:
We can use `padStart` ([safe with node >= 8.2](https://node.green/#ES2017-features-String-padding-String-prototype-padStart))

```suggestion
  const paddedVersion = String(version).padStart(paddingLength, "0");
```